src/main/java/com/savvato/tribeapp/services/ToBeReviewedCheckerServiceImpl.java

```diff
@@ -5,110 +5,127 @@
-        if ((expectedPartOfSpeech == "noun" || expectedPartOfSpeech == "verb") && word.length() == 0) {
-            Exception e = new IllegalStateException();
-            LOGGER.log(Level.SEVERE, "Critical error: expected component " + expectedPartOfSpeech + " is empty!", e);
-            return false;
-        }
-        Boolean validPartOfSpeech = checkPartOfSpeech(word, expectedPartOfSpeech, getWordDetails(word));
-        try {
-            if (validPartOfSpeech) {
-                return true;
-            } else {
-                LOGGER.warning("The word passed in isn't a " + expectedPartOfSpeech + "!");
-                return false;
-            }
-        } catch (Exception e) {
-            LOGGER.warning("The " + expectedPartOfSpeech + " passed in isn't an English word!");
+
+        Boolean validPartOfSpeech = checkPartOfSpeech(word, expectedPartOfSpeech);
+
+        if (validPartOfSpeech) {
+            return true;
+        } else {
+            log.warn("The word passed in isn't a " + expectedPartOfSpeech + "!");
             return false;
         }
+
     }
 
     @Override
     public void validatePhrase(ToBeReviewed tbr) {
-        Optional<RejectedPhrase> matchingRejectedPhrase = rejectedPhraseRepository.findByRejectedPhrase(tbr.toString());
-        if (matchingRejectedPhrase.isEmpty()) {
-            boolean validPhrase = validatePhraseComponent(tbr.getNoun(), "noun") && validatePhraseComponent(tbr.getVerb(), "verb") && validatePhraseComponent(tbr.getAdverb(), "adverb") && validatePhraseComponent(tbr.getPreposition(), "preposition");
-            if (validPhrase) {
-                toBeReviewedRepository.setHasBeenGroomedTrue(tbr.getId());
-            } else {
-                LOGGER.warning("Phrase is invalid.");
-                updateTables(tbr);
-            }
-        }
-        else {
-            LOGGER.warning("Phrase has already been rejected!");
+
+        boolean validPhrase = validatePhraseComponent(tbr.getNoun(), "noun")
+                && validatePhraseComponent(tbr.getVerb(), "verb")
+                && (tbr.getAdverb().equals(Constants.NULL_VALUE_WORD) || validatePhraseComponent(tbr.getAdverb(), "adverb"))
+                && (tbr.getPreposition().equals(Constants.NULL_VALUE_WORD) || validatePhraseComponent(tbr.getPreposition(), "preposition"));
+
+        if (validPhrase) {
+            tbr.setHasBeenGroomed(true);
+            toBeReviewedRepository.save(tbr);
+        } else {
+            log.warn("Phrase is invalid.");
             updateTables(tbr);
         }
     }
 
     @Override
     public void updateTables(ToBeReviewed tbr) {
         rejectedPhraseRepository.save(new RejectedPhrase(tbr.toString()));
-        reviewSubmittingUserRepository.deleteByToBeReviewedId(tbr.getId());
+        ReviewSubmittingUser rsu = new ReviewSubmittingUser(reviewSubmittingUserRepository.findUserIdByToBeReviewedId(tbr.getId()),tbr.getId());
+        reviewSubmittingUserRepository.delete(rsu);
```

Comment:
Would deleteByToBeReviewedId() work here? That way, you wouldn't need line 127 and that call to get the user ID.. 

A delete by toBeReviewedId would delete this entry from the table for all users who submitted this phrase. I believe that is what we want, because we have just determined that this phrase is invalid. 

We do want some way of notifying each of those users that their phrase has been rejected. If you would, please add a TODO comment to the code to that affect, and please create an issue in Jira. Perhaps do those the other way around, and reference the Jira number in the TODO.

Reply:
The original code used a custom deletByToBeReviewedId. I kept getting the following errors:
- could not extract ResultSet; nested exception is org.hibernate.exception.GenericJDBCException: could not extract ResultSet
- Can not issue data manipulation statements with executeQuery()

I fixed it by calling using a ReviewSubmittingUser object and the hibernate .delete() method. I had to troubleshoot this for quite a while and blamed it on the fact that this database table uses a composite primary key. I am not sure why it didn't show up in earlier development, but I did have a cascading process of finding an error, fixing it and uncovering another. I would be happy to troubleshoot this in our meeting, if you want me to make the custom delete query work.

I will add the todo comment and ticket

Reply:
I was wrong. I should have looked at the table. This is the correct way to do it. Never mind this comment.